packages/headless/src/controllers/insight/generated-answer/headless-insight-generated-answer.ts

```diff
@@ -0,0 +1,66 @@
+import {GeneratedAnswerCitation} from '../../../api/generated-answer/generated-answer-event-payload';
+import {updateResponseFormat} from '../../../features/generated-answer/generated-answer-actions';
+import {generatedAnswerInsightAnalyticsClient} from '../../../features/generated-answer/generated-answer-insight-analytics-actions';
+import {GeneratedAnswerState} from '../../../features/generated-answer/generated-answer-state';
+import {GeneratedResponseFormat} from '../../../features/generated-answer/generated-response-format';
+import {executeSearch} from '../../../features/insight-search/insight-search-actions';
+import {InsightEngine} from '../../../insight.index';
+import {
+  GeneratedAnswer,
+  GeneratedAnswerProps,
+  buildCoreGeneratedAnswer,
+} from '../../core/generated-answer/headless-core-generated-answer';
+
+export type {
+  GeneratedAnswerCitation,
+  GeneratedResponseFormat,
+  GeneratedAnswerState,
+  GeneratedAnswer,
+  GeneratedAnswerProps,
+};
+
+/**
+ * Creates an insight `GeneratedAnswer` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `GeneratedAnswer` properties.
+ * @returns A `GeneratedAnswer` controller instance.
+ */
+export function buildGeneratedAnswer(
+  engine: InsightEngine,
+  props: GeneratedAnswerProps = {}
+): GeneratedAnswer {
+  const {dispatch} = engine;
+  const controller = buildCoreGeneratedAnswer(
+    engine,
+    generatedAnswerInsightAnalyticsClient,
+    props
+  );
+
+  return {
+    ...controller,
+
+    get state() {
+      return controller.state;
+    },
+
+    retry() {
```

Comment:
we are leaving the `retry()` & the `rephrase()` methods in the individual controllers and not the core one because of the different analyticsClients right?

Reply:
It's because of the 'executeSearch' function which is specific to the use case here.

Reply:
Should we give it the executeSearch action too and it could define all the functions?

Reply:
I tried that and I found that it's a bit tricky to get that to work with passing the `executeSearch` action, mainly because of the following: https://github.com/coveo/ui-kit/pull/3489#discussion_r1434947111